fem/gridfunc.cpp

```diff
@@ -1790,8 +1768,8 @@ void GridFunction::ProjectGridFunction(const GridFunction &src)
    {
       // Assuming that the projection matrix is the same for all elements
       sameP = true;
-      fes->GetFE(0)->Project(*src.fes->GetFE(0),
-                             *mesh->GetElementTransformation(0), P);
+      fes->GetTypicalFE()->Project(*src.fes->GetTypicalFE(),
+                                   *mesh->GetTypicalElementTransformation(), P);
```

Comment:
Here we don't need to use any of the `GetTypical*` methods since the check `mesh->GetNumGeometries(mesh->Dimension()) == 1` guarantees that we have at least one element locally. And even before that, we have the early exit for empty meshes: `if (!mesh->GetNE()) { return; }`.

Reply:
True, but it's not clear to me that `GetFE(0)` is clearer (in terms of communicating intent) than `GetTypicalFE()`. The result will be the same. I'm OK with either one.

Reply:
I guess my main point here was that we don't actually need `GetTypicalElementTransformation` here -- this is one place where `GetTypicalElementTransformation` is not used in conjunction with `GetRule`. Since we are not going to remove `GetTypicalElementTransformation`, there's no need to revert the change, if you prefer to keep it this way -- as you said the behavior is the same.